src/OMSimulatorLib/ComponentFMUCS.cpp

```diff
@@ -379,6 +384,12 @@ oms_status_enu_t oms::ComponentFMUCS::initializeDependencyGraph_initialUnknowns(
     {
       for (size_t j = startIndex[i]; j < startIndex[i + 1]; j++)
       {
+        if (dependency[j] <= 0 || allVariables.size() < dependency[j])
+        {
+          logError("Out of range of vector allVariables.");
+          logInfo("Dependecies from modelDescription.xml could be wrong. Use flag --ignoreInitialUnknowns=true to ignore dependencies.");
+          return oms_status_fatal;
+        }
```

Comment:
What's about `initializeDependencyGraph_outputs`?

Reply:
Yes, I should add this there as well.

Reply:
Done